src/EthernetConnectionManager.h

```diff
@@ -0,0 +1,187 @@
+#include "ConnectionManager.h"
+
+#include <Ethernet.h>
+#define BOARD_HAS_ETHERNET
+
+class EthConnectionManager : public ConnectionManager {
+public:
+  EthConnectionManager(uint8_t *mac, int ss_pin);
+
+  virtual unsigned long getTime();
+  virtual void init();
+  virtual void check();
+  virtual Client &getClient() { return ethClient; };
+  virtual UDP &getUDP() { return udp; };
+
+private:
+
+  void changeConnectionState(NetworkConnectionState _newState);
+  const int CHECK_INTERVAL_IDLE = 100;
+  const int CHECK_INTERVAL_INIT = 100;
+  const int CHECK_INTERVAL_CONNECTING = 500;
+  const int CHECK_INTERVAL_GETTIME = 100;
+  const int CHECK_INTERVAL_CONNECTED = 10000;
+  const int CHECK_INTERVAL_RETRYING = 5000;
+  const int CHECK_INTERVAL_DISCONNECTED = 1000;
+  const int CHECK_INTERVAL_ERROR = 500;
+
+  unsigned long lastConnectionTickTime, lastNetworkStep;
+  uint8_t* mac;
+  int ss_pin;
+  EthernetClient ethClient;
+  EthernetUDP udp;
+  int connectionTickTimeInterval;
+};
+
+#if !defined(BOARD_HAS_WIFI) && !defined(BOARD_HAS_GSM)
+static const unsigned long NETWORK_CONNECTION_INTERVAL = 30000;
+#endif
+
+EthConnectionManager::EthConnectionManager(uint8_t *mac, int ss_pin = -1) :
+  mac(mac),
+  ss_pin(ss_pin),
+  lastConnectionTickTime(millis()),
+  connectionTickTimeInterval(CHECK_INTERVAL_IDLE) {
+}
+
+unsigned long EthConnectionManager::getTime() {
+  //handled by fallback manager
+  return lastValidTimestamp + 1;
+}
+
+void EthConnectionManager::init() {
+}
+
+void EthConnectionManager::changeConnectionState(NetworkConnectionState _newState) {
+  netConnectionState = _newState;
+  int newInterval = CHECK_INTERVAL_IDLE;
+  switch (_newState) {
+    case CONNECTION_STATE_INIT:
+      newInterval = CHECK_INTERVAL_INIT;
+      break;
+    case CONNECTION_STATE_CONNECTING:
+      newInterval = CHECK_INTERVAL_CONNECTING;
+      break;
+    case CONNECTION_STATE_GETTIME:
+      newInterval = CHECK_INTERVAL_GETTIME;
+      break;
+    case CONNECTION_STATE_CONNECTED:
+      newInterval = CHECK_INTERVAL_CONNECTED;
+      break;
+    case CONNECTION_STATE_DISCONNECTED:
+      newInterval = CHECK_INTERVAL_DISCONNECTED;
+
+      break;
+  }
+  connectionTickTimeInterval = newInterval;
+  lastConnectionTickTime = millis();
+}
+
+void EthConnectionManager::check() {
+  char msgBuffer[120];
+  unsigned long now = millis();
```

Comment:
`now` is used **read-only** within this function so it might as well be **const**.
```suggestion
  unsigned long const now = millis();
```